### PR TITLE
Account registration emails and legal sig emails copied to support

### PR DESCRIPTION
**Pull request by ptrelease**
## Proposed changes

Sends a copy of the account confirmation email to a support team.

## Further comments

Awaiting tests

### CODE REVIEW

**Comment by Gubbsy** [comment]
Note before deploying the Template ID will need updating, along with the legal sigs notify template. We will also need to update the env vars

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "solargraph.autoformat": true
+}
```
Comment (ptrelease) [inline .vscode/settings.json:3]: Remove this file

Comment (Gubbsy) [reply .vscode/settings.json:3]: Ah, not sure how that got in tbh, that's my vscode config

File: app/mailers/notify_mailer.rb [hunk 2]
```diff
@@ -14,6 +26,28 @@ def confirmation_instructions(record, token, opts = {})
...
+  # @param record [User] An instance of User, created by the Devise gem.
+  def confirmation_instructions_copy(record) 
+    template_mail('a44293b7-7263-42b4-8905-44bbedaf1dfa',
+      to: Rails.configuration.x.no_reply_email_address,
+      reply_to_id: Rails.configuration.x.reply_email_guid,
+      personalisation: {
+          confirmation_url: devise_confirmation_url(record, confirmation_token: record.confirmation_token)
+      }
+    )  
+  end
```
Comment (ptrelease) [inline app/mailers/notify_mailer.rb:44]: Need to add a way that Support can identify who the mail was intended for.  Could reuse the same principle that you used for legal agreements - including the email in the subject.

File: app/mailers/notify_mailer.rb [hunk 3]
```diff
@@ -134,19 +168,21 @@ def legal_signatory_agreement_link(
...
     project_title,
     project_reference_number,
-    organisation_name
+    organisation_name, 
+    fao_email
   )
 
     template_mail(
-      'a990844f-700a-4185-9b77-6c1f33cf938e',
+      '9ba83d4a-e445-4c12-9f03-545bcbca4878',
```
Comment (ptrelease) [inline app/mailers/notify_mailer.rb:176]: This is still the test template.  The genuine template still doesn't have the subject suffix added.

File: config/webpacker.yml [hunk 4]
```diff
@@ -53,7 +53,7 @@ development:
   compile: true
 
   # Verifies that correct packages and versions are installed by inspecting package.json, yarn.lock, and node_modules
-  check_yarn_integrity: true
+  check_yarn_integrity: false
```
Comment (ptrelease) [inline config/webpacker.yml:56]: I forget why you disabled this?

File: spec/controllers/legal_agreements/confirm_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      expect(subject).to receive(:send_legal_signatory_email).with("no_reply@test.com", funding_application, "http://test_legal_sig_url.com", " FAO - joe@bloggs.com" )
+
+      # Send email to legal sig 2 and support 
+      expect(subject).to receive(:send_legal_signatory_email).with("john@dear.com", funding_application, "http://test_legal_sig_url.com", "." )
+      expect(subject).to receive(:send_legal_signatory_email).with("no_reply@test.com", funding_application, "http://test_legal_sig_url.com", " FAO - john@dear.com" )
+
+      subject.trigger_legal_signatory_emails(funding_application, subject.current_user)
+    end
+  end
+end
```
Comment (ptrelease) [inline spec/controllers/legal_agreements/confirm_controller_spec.rb:75]: add newline

File: spec/mailers/notify_mailer_spec.rb [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+      organisation_name = "TestOrg"
+
+
+    it "should call the legal_signatory_agreement_link method which " \
+      "send a legal agreements magic link email to user" do
+
+      # check that when legal_signatory_agreement_link is called that
+      # template_mail is called with the expected parameters
+      expect(subject).to receive(:template_mail).with(
+        "9ba83d4a-e445-4c12-9f03-545bcbca4878",
```
Comment (ptrelease) [inline spec/mailers/notify_mailer_spec.rb:62]: wrong template

File: spec/mailers/notify_mailer_spec.rb [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+        project_reference_number, 
+        organisation_name,
+        "."
+      )
+
+    end
+
+  end
+
+end
```
Comment (ptrelease) [inline spec/mailers/notify_mailer_spec.rb:92]: newline

File: spec/controllers/legal_agreements/confirm_controller_spec.rb [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+        project: nil
+      )
+    }
+
+    it "it calls trigger_legal_signatory_emails and where both legal" \
+     "signatories are not the applicant emails are sent to both the legal sigs and support" do
+
+      # Generate magic link for emails
+      expect(subject).to receive(:build_agreement_link).with(anything(), anything()).and_return("http://test_legal_sig_url.com")
+      expect(subject).to receive(:build_agreement_link).with(anything(), anything()).and_return("http://test_legal_sig_url.com")
```
Comment (ptrelease) [inline spec/controllers/legal_agreements/confirm_controller_spec.rb:62]: Is this meant to be here twice?  You can check to see if a function is run twice.

Comment (Gubbsy) [reply spec/controllers/legal_agreements/confirm_controller_spec.rb:62]: I use expect to assert that it is called twice AND mock the reply that we get back to use in the subsequent methods.

Comment (ptrelease) [reply spec/controllers/legal_agreements/confirm_controller_spec.rb:62]: @Gubbsy This is probably the way to go https://rubydoc.info/gems/rspec-mocks/RSpec%2FMocks%2FMessageExpectation:twice

Comment (ptrelease) [reply spec/controllers/legal_agreements/confirm_controller_spec.rb:62]: Unless the return doesn't work then - in which case stick with what you have.

Comment (Gubbsy) [reply spec/controllers/legal_agreements/confirm_controller_spec.rb]: Amened, didn't spot this way of expecting

File: spec/mailers/notify_mailer_spec.rb (hunk 6 above)
Comment (ptrelease) [reply spec/mailers/notify_mailer_spec.rb:65]: Keeping this template and retiring old one.

File: app/mailers/notify_mailer.rb (hunk 3 above)
Comment (ptrelease) [reply app/mailers/notify_mailer.rb:178]: Will rename the test template - retire the previous template

**Pull request merged**